src/v2/providers/https.ts

```diff
@@ -219,7 +219,7 @@ export function onRequest(
     opts = optsOrHandler as HttpsOptions;
   }
 
-  if (isDebugFeatureEnabled('enableCors') || 'cors' in opts) {
+  if (opts.cors !== false && (isDebugFeatureEnabled('enableCors') || 'cors' in opts)) {
```

Comment:
oof lots conditions here that's making my head spin 🤯 

What if we do this in longer form:

```suggestion
  if (isDebugFeatureEnabled('enableCors') || 'cors' in opts) {
    let origin = opts.cors;
    if (isDebugFeatureEnabled('enableCors')) {
      // Respect `cors: false` to turn off cors even if debug feature is enabled.
      origin = opts.cors === false ? false : true 
    }
  }
```